fk/secretreceive.go

```diff
@@ -202,13 +230,77 @@ func countDigits(i int) (count int) {
 	return len(iString)
 }
 
+func getFileNameToWrite(filename string) (string, error) {
+	userDir, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	downloadsDir := userDir + "/Downloads"
+	fileInfo, err := os.Stat(downloadsDir)
+	if err != nil {
```

Comment:
this would be nicely factored out into e.g.
````
err := ensureDirectoryExists(downloadsDir)
if err != nil {
   ...
}
```

Reply:
👍